pgjdbc/src/main/java/org/postgresql/jdbc/Arrays.java

```diff
@@ -0,0 +1,1412 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import org.postgresql.core.BaseConnection;
+import org.postgresql.core.Encoding;
+import org.postgresql.core.Oid;
+import org.postgresql.util.ByteConverter;
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility for using arrays in requests.
+ *
+ * <p>
+ * Binary format:
+ * <ul>
+ * <li>4 bytes with number of dimensions</li>
+ * <li>4 bytes, boolean indicating nulls present or not</li>
+ * <li>4 bytes type oid</li>
+ * <li>8 bytes describing the length of each dimension (repeated for each dimension)</li>
```

Comment:
little nit pick here this is actually 4 bytes of the dimension size and 4 bytes of the lower bound size

Reply:
The following 2 lines break that out.